app/src/main/java/com/ubergeek42/WeechatAndroid/utils/History.kt

```diff
@@ -0,0 +1,54 @@
+package com.ubergeek42.WeechatAndroid.utils
+
+import android.os.Bundle
+import android.text.Editable
+import android.widget.EditText
+import androidx.core.os.bundleOf
+import com.ubergeek42.WeechatAndroid.service.P
+
+class History {
+    private var index = -1
+    private var userInput: Editable? = null
+    private var selectionStart: Int? = null
+    private var selectionEnd: Int? = null
+
+    fun navigateOffset(editText: EditText, offset: Int) {
+        if (index == -1) {
+            userInput = editText.text
+            selectionStart = editText.selectionStart
+            selectionEnd = editText.selectionEnd
+        }
+        val newIndex = index + offset
+        messageAt(newIndex)?.let {
+            editText.text = it
+            editText.post {
```

Comment:
Why post?

Reply:
> Finally, in #553 the user asked about history navigation with Up/Down arrow keys, NOT volume buttons,

Oh, missed that indeed, sorry!

Reply:
Okay, but why `.post {}` though?

Reply:
I initially thought it was necessary to have an effect right after updating the input text, but turns out this is wrong (I mis-diagnosed this while developing for reasons that are annoying to explain).

Removed `post`.